src/loaders/vorbis.c

```diff
@@ -3707,8 +3710,16 @@ static int start_decoder(vorb *f)
    f->comment_list = NULL;
    if (f->comment_list_length > 0)
    {
-      f->comment_list = (char**) setup_malloc(f, sizeof(char*) * (f->comment_list_length));
-      if (f->comment_list == NULL)                  return error(f, VORBIS_outofmem);
+      if (INT_MAX / sizeof(char*) < f->comment_list_length)
+          goto no_comment;
```

Comment:
This is a pointless spaghetti usage of `goto` that saves one entire line of code...